protected/assets/js/patient.js

```diff
@@ -37,6 +37,8 @@ function removeSelectedReferredto(){
   $('#no_referred_to_result').hide();
   $('.selected_referred_to span.name').text('');
   $('#selected_referred_to_wrapper').hide();
+  $('#PatientUserReferral_user_id').val('-1');
```

Comment:
Not sure how I feel about having a magic number in use here. Are we not just able to check if it is null/empty?

Reply:
We need a way to flag when a record should be delete.
If user select a "referred to" contact and remove it before saving, the "selected_referred_to_wrapper" is hide but the value is still there, we just cannot see them.